server/src/test/java/org/opensearch/search/SearchServiceTests.java

```diff
@@ -1413,6 +1414,142 @@ public void testConcurrentSegmentSearchSearchContext() throws IOException {
             .get();
     }
 
+
+    /**
+     * Tests that the slice count is calculated correctly when concurrent search is enabled
+     *  If concurrent search enabled -
+     *       pick index level slice count setting if index level setting is set
+     *       else pick default cluster level slice count setting
+     * @throws IOException
+     */
+    public void testConcurrentSegmentSearchSliceCount() throws IOException {
+        Boolean[][] scenarios = {
+            // cluster setting, index setting
+            {true, null, true},
+            {null, true, true},
+            {false, true, true},
+            {true, false, false},
+            {false, false, false},
+        };
```

Comment:
It looks like you're testing concurrent search disabled case too, do we really need that?

Reply:
Why don't we have it as a {int, int, int} where it's cluster setting, index setting, expected value? And we can just use `null` to indicate not setting it.